src/jit/optimizer.cpp

```diff
@@ -7837,7 +7837,9 @@ GenTree *           Compiler::optIsBoolCond(GenTree *   condBranch,
     if  (opr2->gtOper != GT_CNS_INT)
         return  NULL;
 
-    if  (((unsigned) opr2->gtIntCon.gtIconVal) > 1)
+    ssize_t ival2 = opr2->gtIntCon.gtIconVal;
+
+    if (ival2 != 0 && ival2 != 1)
         return NULL;
```

Comment:
Thanks for the suggestion. I did consider this option of adding a property, however decided against it. Reading: (ival2 != 0 && ival2 != 1) makes intent pretty explicit already, and for all but for a few specific value pairs around signularities, using special case property methods breaks down for most general cases and have to use an inline sequence against literal constants anyway.